tests/expectations/tests/no-recursive-whitelisting.rs

```diff
@@ -25,6 +25,3 @@ fn bindgen_test_layout_Foo() {
 impl Clone for Foo {
     fn clone(&self) -> Self { *self }
 }
-impl Default for Foo {
-    fn default() -> Self { unsafe { ::std::mem::zeroed() } }
-}
```

Comment:
I print out what I get from ctx.whitelisted_items() and Foo is in it. I think this is correct from the current algorithm.